crates/protogen/src/metastore/types/catalog.rs

```diff
@@ -10,19 +10,28 @@ use super::options::{
     CredentialsOptions,
     DatabaseOptions,
     InternalColumnDefinition,
-    TableOptions,
     TableOptionsInternal,
+    TableOptionsV0,
+    TableOptionsV1,
     TunnelOptions,
 };
 use crate::gen::common::arrow::ArrowType;
 use crate::gen::metastore::catalog::{self, type_signature};
 use crate::{FromOptionalField, ProtoConvError};
 
+/// The current version of the catalog.
+/// this is incremented every time there is a breaking change to the catalog.
+pub const CURRENT_CATALOG_VERSION: u32 = 1;
+
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct CatalogState {
+    /// the version of the catalog state. This increments on every Mutation
     pub version: u64,
     pub entries: HashMap<u32, CatalogEntry>,
     pub deployment: DeploymentMetadata,
+    /// This is the version of the catalog implementation.
+    /// This is incremented when the catalog changes in a way that is not backwards compatible.
```

Comment:
In these comments it might be useful to clarify that it's the physical representation of the catalog that we're ensuring backwards compatibility, not semantic stuff.